### ISSUE TITLE
SEO: Remove URL suffix ID

### ISSUE THREAD

**Issue by fczaja** [odoo/odoo#12530]
## Impacted versions
- 9.0
## Goal

The ID should be removed from the URL.
## Reason

The URL is one of the most important SEO criteria.
Unimportant things should not stand in the URL, so that a page is relevant for important searches.
An ID can distort the meaning of the URL.
The URL should not be unnecessarily long, so they can be seen complete on the search results page of Google.
## Current behavior

the ID moment is appended to a URL:
shop/product/imac-11
shop/category/computers-12
blog/our-blog-1/post/the-future-of-emails-1
## Expected behavior

In the URL is no ID:
shop/product/imac
## This information is available

https://github.com/odoo/odoo/issues/12527

**Comment by zeroheure** [comment]
I fully disagree. It is useful to have ID in the URL for SEO.
When migrating from another ecommerce system to a new one, it is usual to build 301 redirections in order to preserve SEO. But you will soon get a bunch of them, at least one per product and often one per image of product. It is a pain to build, and it can slow down the web server.

There is a far easier and quicker strategy. With IDs in the URL one can forget all 301 for products and related images :
1. First, prepare the migration by adding IDs in URLs at the old ecommerce system and request the shop engine to do automatic redirections. In a few weeks the new URLs with IDs are registered in major search engines. 
1. Second, when you open your new Odoo web store, as long as you keep the same IDs (those are the pKey of product_template) a simple regex does the job to redirect old URLs with IDs to Odoo URLs with IDs.

In conclusion, keep IDs in the URL, at least optionaly.

**Comment by pedrobaeza** [comment]
I also don't agree with this. It's not only useful, but needed in the way Odoo works, and this doesn't affect SEO by all the sources I know.
